### ISSUE TITLE
Gradient overflow occurs while training tnt-ti model

### ISSUE THREAD

**Issue by jimmyflycv** [huawei-noah/CV-Backbones#79]
^@^@Train: 41 [   0/625 (  0%)]  Loss:  4.564162 (4.5642)  Time: 96.744s,   21.17/s  (96.744s,   21.17/s)  LR: 8.284e-04  Data: 94.025 (94.025)
^@^@^@^@Train: 41 [  50/625 (  8%)]  Loss:  4.395192 (4.4797)  Time: 2.742s,  746.96/s  (7.383s,  277.38/s)  LR: 8.284e-04  Data: 0.057 (4.683)
^@^@^@^@Train: 41 [ 100/625 ( 16%)]  Loss:  4.424296 (4.4612)  Time: 2.741s,  747.15/s  (6.529s,  313.66/s)  LR: 8.284e-04  Data: 0.056 (3.831)
Gradient overflow.  Skipping step, loss scaler 0 reducing loss scale to 16384.0
Gradient overflow.  Skipping step, loss scaler 0 reducing loss scale to 16384.0
Gradient overflow.  Skipping step, loss scaler 0 reducing loss scale to 16384.0
Gradient overflow.  Skipping step, loss scaler 0 reducing loss scale to 16384.0

And the top-1 acc is only 0.2 after 40 epochs.

Any tips available here, dear @iamhankai @yitongh

**Comment by iamhankai** [comment]
The log looks normal. Please go ahead and train the model to the end.

**Comment by jimmyflycv** [comment]
ok.

**Comment by jimmyflycv** [comment]
After 84/160 epochs training, top1 acc was only 5.46% and it increased very very slowly, i guess there might be mistake here.

Test: [   0/24]  Time: 16.759 (16.759)  Loss:  1.0276 (1.0276)  Acc@1: 75.5371 (75.5371)  Acc@5: 93.6035 (93.6035)
Test: [  24/24]  Time: 0.532 (2.110)  Loss:  0.8602 (1.4612)  Acc@1: 80.8962 (66.7000)  Acc@5: 93.8679 (87.7400)
Test (EMA): [   0/24]  Time: 16.192 (16.192)  Loss:  6.0643 (6.0643)  Acc@1:  3.2227 ( 3.2227)  Acc@5:  9.6191 ( 9.6191)
Test (EMA): [  24/24]  Time: 0.552 (2.107)  Loss:  5.8538 (5.9276)  Acc@1:  4.1274 ( 5.4660)  Acc@5: 11.4387 (14.2580)
Current checkpoints:
 ('./saved_models/train/20220111-212638-ptnt_ti_patch16_192-192/checkpoint-83.pth.tar', 5.465999999084473)
 ('./saved_models/train/20220111-212638-ptnt_ti_patch16_192-192/checkpoint-82.pth.tar', 5.215999999084473)
 ('./saved_models/train/20220111-212638-ptnt_ti_patch16_192-192/checkpoint-81.pth.tar', 4.8479999989318845)
 ('./saved_models/train/20220111-212638-ptnt_ti_patch16_192-192/checkpoint-80.pth.tar', 4.6160000002288815)
 ('./saved_models/train/20220111-212638-ptnt_ti_patch16_192-192/checkpoint-79.pth.tar', 4.286000001373291)
 ('./saved_models/train/20220111-212638-ptnt_ti_patch16_192-192/checkpoint-78.pth.tar', 3.999999999923706)
 ('./saved_models/train/20220111-212638-ptnt_ti_patch16_192-192/checkpoint-77.pth.tar', 3.651999998474121)
 ('./saved_models/train/20220111-212638-ptnt_ti_patch16_192-192/checkpoint-76.pth.tar', 3.3299999997711183)
 ('./saved_models/train/20220111-212638-ptnt_ti_patch16_192-192/checkpoint-75.pth.tar', 3.0259999996185303)
 ('./saved_models/train/20220111-212638-ptnt_ti_patch16_192-192/checkpoint-74.pth.tar', 2.7759999994659426)

My training script is below, and i wonder if the ti mdel could be trained from scratch?
python -m torch.distributed.launch --nproc_per_node=4  train.py  data_path --model ptnt_ti_patch16_192 --sched cosine --epochs 150 --opt adamw -j 8 --warmup-lr 5e-7 --mixup .8 --cutmix 1.0 --model-ema --model-ema-decay 0.99996 --aa rand-m9-mstd0.5-inc1 --color-jitter 0.4 --warmup-epochs 10 --opt-eps 1e-8 --repeated-aug --remode pixel --reprob 0.25 --amp --lr 5e-4 --weight-decay .05 --drop 0 --drop-path .1 -b 512 --output  ./saved_models/

**Comment by jimmyflycv** [comment]
Train: 103 [ 550/625 ( 88%)]  Loss:  3.068429 (3.8811)  Time: 2.817s,  726.90/s  (5.927s,  345.55/s)  LR: 2.311e-04  Data: 0.056 (3.224)
Train: 103 [ 600/625 ( 96%)]  Loss:  4.100181 (3.8980)  Time: 7.432s,  275.56/s  (5.949s,  344.26/s)  LR: 2.311e-04  Data: 4.745 (3.246)
Train: 103 [ 624/625 (100%)]  Loss:  3.959771 (3.9024)  Time: 3.197s,  640.66/s  (5.872s,  348.75/s)  LR: 2.311e-04  Data: 0.520 (3.169)
Test: [   0/24]  Time: 16.471 (16.471)  Loss:  0.9927 (0.9927)  Acc@1: 78.8086 (78.8086)  Acc@5: 94.0918 (94.0918)
Test: [  24/24]  Time: 0.513 (2.109)  Loss:  0.8202 (1.3808)  Acc@1: 83.6085 (69.2780)  Acc@5: 94.6934 (89.3980)
Test (EMA): [   0/24]  Time: 15.783 (15.783)  Loss:  5.4779 (5.4779)  Acc@1:  9.0332 ( 9.0332)  Acc@5: 20.0684 (20.0684)
Test (EMA): [  24/24]  Time: 0.530 (2.069)  Loss:  4.9858 (5.4163)  Acc@1: 11.9104 (10.2000)  Acc@5: 27.5943 (23.3300)
Current checkpoints:
 ('./saved_models/train/20220111-212638-ptnt_ti_patch16_192-192/checkpoint-103.pth.tar', 10.200000018615723)
 ('./saved_models/train/20220111-212638-ptnt_ti_patch16_192-192/checkpoint-102.pth.tar', 9.88200000213623)
 ('./saved_models/train/20220111-212638-ptnt_ti_patch16_192-192/checkpoint-101.pth.tar', 9.612000004577636)
 ('./saved_models/train/20220111-212638-ptnt_ti_patch16_192-192/checkpoint-100.pth.tar', 9.336000004272462)
 ('./saved_models/train/20220111-212638-ptnt_ti_patch16_192-192/checkpoint-99.pth.tar', 9.094000006713868)
 ('./saved_models/train/20220111-212638-ptnt_ti_patch16_192-192/checkpoint-98.pth.tar', 8.867999995727539)
 ('./saved_models/train/20220111-212638-ptnt_ti_patch16_192-192/checkpoint-97.pth.tar', 8.674000000915527)
 ('./saved_models/train/20220111-212638-ptnt_ti_patch16_192-192/checkpoint-96.pth.tar', 8.43599999267578)
 ('./saved_models/train/20220111-212638-ptnt_ti_patch16_192-192/checkpoint-95.pth.tar', 8.251999998168944)
 ('./saved_models/train/20220111-212638-ptnt_ti_patch16_192-192/checkpoint-94.pth.tar', 8.080000008850098)

**Comment by iamhankai** [comment]
The top-1 acc without EMA already achieves 69.2780. It's normal.

**Comment by jimmyflycv** [comment]
> The top-1 acc without EMA already achieves 69.2780. It's normal.

Thanks very much for ur answer
